onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -251,7 +260,8 @@ Status QkvToContext(
   T* v = k + elements_k;
 
   // For fused TRT attention, qkv need transpose to BxSxNx3xH
-  bool use_fused_kernel = (nullptr != fused_runner && data.bias != nullptr);
+  bool use_fused_kernel = (nullptr != fused_runner && data.bias != nullptr && !parameters.is_unidirectional);
+  bool use_fused_causal = (nullptr != fused_runner && data.bias != nullptr && parameters.is_unidirectional);
 
   if (nullptr != data.gemm_buffer) {
     if (data.bias == nullptr) {
```

Comment:
>if (data.bias == nullptr) { [](http://example.com/codeflow?start=4&length=27)

I don't see a case that data.bias can be nullptr. This is for Cross-Attention support?

Reply:
That's for quantized version of attention where bias has been added: https://github.com/microsoft/onnxruntime/blob/1b58331fb342d7db457df1b7a735d23619d51927/onnxruntime/contrib_ops/cuda/quantization/attention_quantization.cc#L195